src/language/sharedLanguage.ts

```diff
@@ -103,17 +118,23 @@ const replaceRootTag = (domNode: any) => {
   }
 };
 
-const replaceParameters = (nameString: string | undefined, params: string[]) => {
+const replaceParameters = (nameString: string | undefined, params: LangParams) => {
   if (nameString === undefined) {
     return nameString;
   }
   let mutatingString = nameString;
-  params.forEach((param: string, index: number) => {
-    mutatingString = mutatingString.replaceAll(`{${index}}`, param);
+  params.forEach((param, index: number) => {
+    if (param !== undefined) {
```

Comment:
Can the params array include undefined? Something like this => ` ['value1', 'value2', undefined, 'value3']` since we have a statement for param !== undefined? 🤔

Reply:
They probably won't be, but when I added more strict typing, some code paths started complaining because they sent in types that were (either explicitly or inferred to be) `(string | undefined)[]`. I just made sure the code worked with it being undefined as well, but I doubt it's in broad use today.